plugins/modules/ec2_instance.py

```
@@ -2579,14 +2603,89 @@ def ensure_present(
+    
+    # List of instance types to try (primary first, then alternates)
+    instance_types_to_try = [primary_instance_type] if primary_instance_type else []
+    if alternate_instance_types:
+        instance_types_to_try.extend(alternate_instance_types)
+    last_error = None
+    
+    for i, instance_type in enumerate(instance_types_to_try):
+        # Create a copy of the instance spec with the current instance type to maintain original values
+        current_spec = copy.deepcopy(instance_spec)
+        current_spec["InstanceType"] = instance_type
+        try:
+            result = run_ec2_instances(client, **current_spec)
+            # If we successfully launched with an alternate instance type, update the module params
+            # to reflect the actually used instance type to prevent later modification attempts
+            if i > 0:
+                module.warn(f"Successfully launched instance with alternate instance type '{instance_type}' "
+                           f"after primary instance type '{primary_instance_type}' failed with InsufficientInstanceCapacity")
+                # Update the module parameters to reflect the actually used instance type
+                module.params["instance_type"] = instance_type
+            
+            return result
+            
+        except AnsibleAWSError as e:
+            except is_ansible_aws_error_message("Invalid IAM Instance Profile ARN"):
+                # Check if this is an IAM instance profile error
+            if is_ansible_aws_error_message("Invalid IAM Instance Profile ARN")(e):
+                # IAM instance profile error - likely due to eventual consistency
+                # If the instance profile has just been created, it takes some time
+                # to be visible by EC2. Wait 10 seconds and retry with the same instance type
+                time.sleep(10)
+            try:
+                result = run_ec2_instances(client, **current_spec)
```

This part of the code is repeated

---

When a new Ec2 is created, the instance profile attachment could take time, So this exception is ignored and ec2 operation is retried. Its repeated because its in the exception block and we want to retry the operation when the exception is for invalid instance profile